lib/cli/interactive-setup/service.js

```diff
@@ -55,27 +58,69 @@ const projectNameInput = async (workingDir) =>
     })
   ).projectName.trim();
 
+const resolveProjectNameInput = async (options, workingDir) => {
+  if (options.name) {
+    if (!isValidServiceName(options.name)) {
+      throw new ServerlessError(INVALID_PROJECT_NAME_MESSAGE, 'INVALID_PROJECT_NAME');
+    }
+
+    let alreadyTaken = false;
+    try {
+      await fs.promises.access(join(workingDir, options.name));
+      alreadyTaken = true;
+    } catch {
+      // Pass
+    }
+
+    if (alreadyTaken) {
+      throw new ServerlessError(
+        `Path ${options.name} is already taken`,
+        'TARGET_FOLDER_ALREADY_EXISTS'
+      );
+    }
+
+    return options.name;
+  }
+
+  return projectNameInput(workingDir);
+};
+
 module.exports = {
   isApplicable({ serviceDir }) {
     return !serviceDir;
```

Comment:
I think we should also throw here errors, if given options were passed but user is in service context (we should not silently ignore them)